### PR TITLE
Bump istio to v1.2.5

### PR DESCRIPTION
**Pull request by guangbochen**
**Issue:**
Bump istio to v1.2.5

**prerequisite:**
push the following images to rancher registry
```
docker pull istio/sidecar_injector:1.2.4
docker tag istio/sidecar_injector:1.2.4 rancher/istio-sidecar_injector:1.2.4
docker push rancher/istio-sidecar_injector:1.2.4

docker pull istio/galley:1.2.4
docker tag istio/galley:1.2.4 rancher/istio-galley:1.2.4
docker push rancher/istio-galley:1.2.4

docker pull istio/mixer:1.2.4
docker tag istio/mixer:1.2.4 rancher/istio-mixer:1.2.4
docker push rancher/istio-mixer:1.2.4

docker pull istio/pilot:1.2.4
docker tag istio/pilot:1.2.4 rancher/istio-pilot:1.2.4
docker push rancher/istio-pilot:1.2.4

docker pull istio/citadel:1.2.4
docker tag istio/citadel:1.2.4 rancher/istio-citadel:1.2.4
docker push rancher/istio-citadel:1.2.4

docker pull istio/kubectl:1.2.4
docker tag istio/kubectl:1.2.4 rancher/istio-kubectl:1.2.4
docker push rancher/istio-kubectl:1.2.4

docker pull istio/proxyv2:1.2.4
docker tag istio/proxyv2:1.2.4 rancher/istio-proxyv2:1.2.4
docker push rancher/istio-proxyv2:1.2.4

docker pull istio/proxy_init:1.2.4
docker tag istio/proxy_init:1.2.4 rancher/istio-proxy_init:1.2.4
docker push rancher/istio-proxy_init:1.2.4

docker pull istio/node-agent-k8s:1.2.4
docker tag istio/node-agent-k8s:1.2.4 rancher/istio-node-agent-k8s:1.2.4
docker push rancher/istio-node-agent-k8s:1.2.4
```

**Related Issue:**
https://github.com/rancher/rancher/issues/21487

### CODE REVIEW

**Comment by loganhz** [comment]
There are still some other changes in charts `https://github.com/istio/istio/commits/1.2.4/install/kubernetes/helm`

We should sync with upstream.

**Comment by guangbochen** [comment]
I have already checked, let me know if you found any particular part is missing, thanks.

**Comment by loganhz** [comment]
https://github.com/istio/istio/commit/4815f576fd29bcfd58e44ff948e295ee7ec66093#diff-6fa1fbfa8139f4a486ee2c3cad75077a
https://github.com/istio/istio/commit/2f86b739adcaa5a26eaec85d984c305178f75e10#diff-6fa1fbfa8139f4a486ee2c3cad75077a
https://github.com/istio/istio/commit/bbe7afd4d157ac105e84b6b9fc9cb695f188c684#diff-6fa1fbfa8139f4a486ee2c3cad75077a
https://github.com/istio/istio/commit/31b3b024fe8e56b871308e626eecbb90e21979c1#diff-6fa1fbfa8139f4a486ee2c3cad75077a
https://github.com/istio/istio/commit/fbd6bf0e6e899ef00c72f9d6254579b468d9222a#diff-6fa1fbfa8139f4a486ee2c3cad75077a
https://github.com/istio/istio/commit/5e948bb58ab9effb3e92cb1b1011d03e8f049982#diff-6fa1fbfa8139f4a486ee2c3cad75077a

**Comment by guangbochen** [comment]
Updated, thanks.

**Comment by loganhz** [comment]
LGTM

**Comment by cjellick** [comment]
going to wait to merge until after our taints/tolerations discussion

**Comment by cjellick** [comment]
Big question: regarding the images pushing: we have processes for scripts that build allow you to pull and push rancher images into your airgap setup.  Do we cover this for istio images for this?

I am not going to pull/push these images manually @cloudnautique and @drpebcak are going to automate.

**Comment by loganhz** [comment]
We have a logic to find the images in system chart. Monitoring is also using this logic. So it's fine.

**Comment by cjellick** [comment]
can you point me to this logic?

**Comment by loganhz** [comment]
@orangedeng ^^ Please point to the code

And find out why istio images is not in the image-text file of `https://github.com/rancher/rancher/releases/tag/v2.3.0-alpha7`

**Comment by orangedeng** [comment]
Here is the code for image generation https://github.com/rancher/rancher/blob/master/scripts/package#L41-L42

**Comment by loganhz** [comment]
So it always scan master branch of system charts for now..`git clone --branch master`

**Comment by drpebcak** [comment]
Mirrored Istio images here: https://github.com/rancher/rancher/pull/22366

**Comment by loganhz** [comment]
@guangbochen Please upgrade to v1.2.5

**Comment by loganhz** [comment]
Please also submit a PR for the images https://github.com/rancher/rancher/blob/master/scripts/istio-images-mirror

**Comment by guangbochen** [comment]
Updated and link the mirror images PR https://github.com/rancher/rancher/pull/22433

File: charts/rancher-istio/0.0.2/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: rancher-istio
 version: 0.0.2
-appVersion: 1.2.0
+appVersion: 1.2.4
```
Comment (loganhz) [inline charts/rancher-istio/0.0.2/Chart.yaml]: should be 1.2.5

Comment (guangbochen) [reply charts/rancher-istio/0.0.2/Chart.yaml]: updated.

**Comment by loganhz** [comment]
lgtm

**Pull request merged**